src/axom/sidre/core/Group.hpp

```diff
@@ -1179,14 +1179,39 @@ class Group
   /*!
    * \brief Load the Group from a file.
    *
-   * \param path      file path
-   * \param protocol  I/O protocol
+   * \param path     file path
+   * \param new_name the group name stored in the file
+   * \param protocol I/O protocol
    * \param preserve_contents   Preserve existing contents of group if true
    */
   void load(const std::string& path,
+            std::string & new_name,
             const std::string& protocol = SIDRE_DEFAULT_PROTOCOL,
             bool preserve_contents = false);
 
+  /*!
+   * \brief Load a file into a new child group.
```

Comment:
Alternately, we could remove `loadChild()` altogether.  It is purely syntactic sugar, and the corner cases of naming are particular enough that we might want to push them back onto the host code.  I think it might be useful and appreciated, though.

Reply:
@kennyweiss , @nselliott , @cyrush , there really wasn't enough time in the last Axom meeting to fully develop `loadChild()`.  It was an idea that we thought would be a common enough use case for host codes to make a convenience routine.

Why don't we simplify it?  Just as with `load()`, naming is the really sticky issue.  Why don't we require the the caller to provide a name for the new Group, just as with `createGroup()`, and return the stored name from the file in an out-variable, just as with `load()`?  This way, Sidre isn't trying to guess renaming policy (custom prefix? zero-filled sequential postfix? don't care except that it's unique?) or empty-string policy.

Reply:
Please review my changes.  Thanks!